tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        let assert_keys = ["valid2"].to_vec();
+
+        // "update-ssh-keys --add one valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_no_replace() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let pubkeyvalid2 = "valid2";
+        let testnameone = "one";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys.clone()),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        // "update-ssh-keys --no-replace --add one valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            false,        // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnameone, assert_keys.clone()),
+            Err(err) => panic!("update_ssh_keys --no-replace --add failed {}", err),
+        }
+
+        // "update-ssh-keys --no-replace --add-force one valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnameone,  // testname
+            true,         // is_force
+            false,        // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --no-replace --add-force failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_add_two() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
```

Comment:
Code below can be replaced by a call to `run_test_add_one_file(&ctx)`

Reply:
Called `run_test_add_one_file(&ctx.ssh_dir)`.